src/taco-livereload/lib/browserSyncServer.js

```diff
@@ -0,0 +1,134 @@
+            rule: {
+                match: /<\/body>/i,
+                fn: function (snippet, match) {
+                    return monkeyPatch() + snippet + match;
+                }
+            }
+        },
+        minify: false,
+        ghostMode: true
+    };
+
+
+    // User provided options take precedence over the default ones
+    self.options = merge(defaultOptions, self.options);
+
+    // Initialize the browser-sync instance
+    self.browserSync.init(self.options, function (err, bs) {
+
+        // Once BrowserSync is ready, resolve the promise
+        if (err) {
+            deferred.reject(err);
+            return;
+        }
+
+        // If the user specified the '--tunnel' option, use a localtunnel.me address
+        var serverUrl = self.options.tunnel ? bs.options.getIn(['urls', 'tunnel']) : bs.options.getIn(['urls', 'external']);
+
+        // In case there is no external url
+        // e.g: When the machine is not connected to any network
+        if (!serverUrl) {
+            // Usually, this err msg ends up being null in this case.
+            var error = new Error('No External URLs available. Make sure your computer is connected to a network.' + (err ? err.msg : ''));
+            deferred.reject(error);
+            return;
+        }
+
+        deferred.resolve(serverUrl);
+        return;
+    });
+
+    return deferred.promise;
+};
+
+// Reloads all the connected browsers
+BrowserSyncServer.prototype.reloadBrowsers = function () {
+    this.browserSync.reload();
+};
+
+// Either reloads the whole file (e.g: html)
+// ... or injects changes into the file (e.g: css)
+// ... the strategy used depends on the file type
+BrowserSyncServer.prototype.reloadFile = function (file) {
+    this.browserSync.reload(file);
+};
+
+// Stops the browsersync server
+BrowserSyncServer.prototype.stopServer = function () {
+    this.browserSync.exit();
+};
+
+
+/**
+ * Private function that adds the code snippet to deal with reloading
+ * files when they are served from platform folders
+ * This is necessary to allow clicks & scrolls (one of BrowserSync's functionalities) to function well across
+ *    different platforms (e.g: IOS and Android)
+ */
+function monkeyPatch() {
+    var script = function () {
+        window.__karma__ = true;
+        (function patch() {
+            if (typeof window.__bs === 'undefined') {
+                window.setTimeout(patch, 500);
+            } else {
+                var oldCanSync = window.__bs.prototype.canSync;
+                window.__bs.prototype.canSync = function (data, optPath) {
+                    data.url = window.location.pathname.substr(0, window.location.pathname.indexOf('/www')) + data.url.substr(data.url.indexOf('/www'));
+                    return oldCanSync.apply(this, [data, optPath]);
+                };
+            }
+        } ());
+    };
+    return '<script>(' + script.toString() + '());</script>';
+}
+
+
+module.exports = BrowserSyncServer;
```

Comment:
Please add a newline at the end of this file and any others that are missing newlines.